Presentation/PresentationStyle.swift

```diff
@@ -99,7 +99,22 @@ public extension PresentationStyle {
             return from.modallyPresentQueued(vc, options: options) {
                 Future { completion in
                     let bag = DisposeBag()
-                    bag += viewController.installDismissButton().onValue { completion(.failure(PresentError.dismissed)) }
+
+                    // The presentationController of an alert controller should not have its delegate modified
+                    if !(vc is UIAlertController) {
+                        let delegate = AdaptiveProxyPresentationDelegate(presentationOptions: options)
+                        bag.hold(delegate)
+
+                        vc.presentationController?.delegate = delegate
+
+                        bag += delegate.didDismissSignal.onValue {
```

Comment:
This can be solved similarly as suggested above:

```swift
vc.didDismissSignal(allowSwipeDismissAlways: options.contains(.allowSwipeDismissAlways)).onValue {
```

Reply:
Since we want to implement all the methods from `UIAdaptivePresentationDelegate` and redirect them to a `customAdaptivePresentationDelegate` it doesn't make sense to do that in a `didDismissSignal`. That would be a good solution if we wanted to implement just one method from the delegate. 

That way it's possible to implement your own `UIAdaptivePresentationDelegate` with the downside of having our own proxy.